docs/docsite/rst/user_guide/playbooks_tags.rst

```diff
@@ -1,227 +1,402 @@
 
-    - block:
-      - name: First task to run
-        ...
-      - name: Second task to run
-        ...
-      tags:
-      - mytag
+* ``--tags all`` - run all tasks, ignore tags (default behavior)
+* ``--tags [tag1, tag2]`` - run only tasks with the tags ``tag1`` and ``tag2``
+* ``--skip-tags [tag3, tag4]`` - run all tasks except those with the tags ``tag3`` and ``tag4``
+* ``--tags tagged`` - run only tasks with at least one tag
+* ``--tags untagged`` - run only tasks with no tags
 
+For example, to run only the "configuration" and "packages" parts of a very long playbook:
 
-.. _special_tags:
+.. code-block:: bash
 
-Special Tags
-````````````
+   ansible-playbook example.yml --tags "configuration,packages"
 
-There is a special ``always`` tag that will always run a task, unless
-specifically skipped (``--skip-tags always``)
+To run all tasks except those tagged ``packages``:
 
-Example::
+.. code-block:: bash
 
-    tasks:
-    - debug:
-        msg: "Always runs"
-      tags:
-      - always
+   ansible-playbook example.yml --skip-tags "packages"
 
-    - debug:
-        msg: "runs when you use tag1"
-      tags:
-      - tag1
+Previewing the results of using tags
+------------------------------------
 
-.. versionadded:: 2.5
+Especially when you run a role or playbook someone else has written or edited, you may not know which tasks have which tags, or which tags exist at all. Ansible offers two command-line flags for :ref:`ansible-playbook` that help you manage tagged playbooks:
+
+* ``--list-tags`` - generate a list of available tags
+* ``--list-tasks`` - when used with ``--tags tagname`` or ``--skip-tags tagname``, generate a preview of tagged tasks
+
+For example, if you do not know whether the tag for configuration tasks is ``config`` or ``conf`` in a playbook, role, or tasks file, you can display all available tags without running any tasks:
+
+.. code-block:: bash
+
+   ansible-playbook example.yml --list-tags
+
+If you do not know which tasks have the tags ``configuration`` and ``packages``, you can pass those tags and add ``--list-tasks``. Ansible lists the tasks but does not execute any of them.
+
+.. code-block:: bash
+
+   ansible-playbook example.yml --tags "configuration,packages" --list-tasks
+
+These command-line flags have one limitation - they cannot show tags or tasks within dynamically included files or roles. See :ref:`dynamic_vs_static` for more information on differences between static imports and dynamic includes.
+
+.. _selective_reuse:
+
+Selectively running tagged tasks in re-usable files
+---------------------------------------------------
+
+If you have a role or a tasks file with tags defined at the task or block level, you can selectively run or skip those tagged tasks in a playbook if you use a dynamic include instead of a static import. You must use the same tag on the included tasks and on the include statement itself. For example if you use a file like this:
```

Comment:
It's usually best to not break sentences up with examples. So maybe say here:

"For example, you might use a file with tagged and untagged tasks:"

See the next two suggestions for additional sentence fragments.